### PR TITLE
feat(measurements) Add measurements fields to transactions and discover datasets.

### PR DESCRIPTION
**Pull request by fpacifici**
Depends on https://github.com/getsentry/snuba/pull/1241

This allows the clients to use individual measurements in the queries or aggregations over individual queries.

Example:
```
selected_columns: ["measurements[lcp]"]
```

Specifically it:
- adds the fields to both the transactions and discover datasets
- fixes the table matching logic in discover

### CODE REVIEW

**Comment by fpacifici** [comment]
Design is on hold. Converting to draft.

**Review by markstory — COMMENTED** [review]
This interface looks good to me. 👍

**Comment by fpacifici** [comment]
Resurrected since the project is moving again.

File: tests/test_discover_api.py [hunk 1]
```diff
@@ -895,3 +899,21 @@ def test_count_null_user_consistency(self):
...
         assert data["data"][0]["uniq_user"] == 0
+
+    def test_individual_measurement(self) -> None:
+        response = self.app.post(
+            "/query",
+            data=json.dumps(
+                {
+                    "dataset": "discover",
+                    "project": self.project_id,
+                    "selected_columns": ["event_id", "measurements[lcp]"],
```
Comment (evanh) [inline tests/test_discover_api.py:910]: I would test `lcp.elementSize` too, to make sure the extra dots don't mess things up.

File: snuba/datasets/discover.py [hunk 2]
```diff
@@ -87,6 +90,13 @@ def match_query_to_table(
         elif transactions_only_columns.get(col.column_name):
             has_transaction_columns = True
 
+    for subscript in query.get_all_ast_referenced_subscripts():
```
[the remark this answers is not in the document]
Comment (evanh) [reply snuba/datasets/discover.py:93]: Wouldn't the `Column` inside the `SubscriptableReference` be surfaced in `get_all_ast_referenced_columns`?

[the remark this answers is not in the document]
Comment (evanh) [reply snuba/datasets/discover.py:93]: Also did you test this?

File: tests/test_discover_api.py (hunk 1 above)
Comment (evanh) [reply tests/test_discover_api.py:910]: Bump, but it's up to you.

**Review by evanh — APPROVED** [review]

**Comment by fpacifici** [comment]
Comments applied and default NULL added

**Review by evanh — APPROVED** [review]

**Pull request merged**